### PR TITLE
migrate play kube to spec gen

### PR DESCRIPTION
**Pull request by baude**
we need to migrate play kube away from using the old container creation
method.  the new approach is specgen and this aligns play kube with
container creation in the rest of podman.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: pkg/specgen/generate/kube/kube.go [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func ToPodGen(ctx context.Context, podName string, podYAML *v1.PodTemplateSpec) (*specgen.PodSpecGenerator, error) {
+	p := specgen.NewPodSpecGenerator()
+	p.Name = podName
+	p.Labels = podYAML.ObjectMeta.Labels
+	// TODO we only configure Process namespace. We also need to account for Host{IPC,Network,PID}
```
Comment (mheon) [inline pkg/specgen/generate/kube/kube.go:23]: For this, we'd remove the namespace from SharedNamespaces and then set it to host on all containers

File: pkg/specgen/generate/kube/kube.go [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+	s.Pod = podID
+
+	setupSecurityContext(s, containerYAML)
+
+	// Since we prefix the container name with pod name to work-around the uniqueness requirement,
+	// the seccomp profile should reference the actual container name from the YAML
+	// but apply to the containers with the prefixed name
+	s.SeccompProfilePath = seccompPaths.FindForContainer(containerYAML.Name)
+
+	var err error
```
Comment (mheon) [inline pkg/specgen/generate/kube/kube.go:72]: This is unnecessary - we have a `:=` on err two lines below

File: pkg/specgen/generate/kube/kube.go [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+	if err != nil {
+		return nil, err
+	}
+	s.WorkDir = "/"
+	if imageData != nil && imageData.Config != nil {
+		if imageData.Config.WorkingDir != "" {
+			s.WorkDir = imageData.Config.WorkingDir
+		}
+		s.Command = imageData.Config.Entrypoint
+		s.Labels = imageData.Config.Labels
```
Comment (mheon) [inline pkg/specgen/generate/kube/kube.go:119]: This bit, WorkingDir, and StopSignal are unnecessary - I know we need to rework Command and Entrypoint for Kube compat, but the rest should be handled by Specgen

File: pkg/specgen/generate/kube/kube.go [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+		if !exists {
+			return nil, errors.Errorf("Volume mount %s specified for container but not configured in volumes", volume.Name)
+		}
+		if err := parse.ValidateVolumeCtrDir(volume.MountPath); err != nil {
+			return nil, errors.Wrapf(err, "error in parsing MountPath")
+		}
+		mount := spec.Mount{
+			Type:        "bind",
+			Destination: hostPath,
+			Source:      volume.MountPath,
```
Comment (mheon) [inline pkg/specgen/generate/kube/kube.go:175]: I think that Options should be unconditionally set to `[]string{"bind"}` here

File: pkg/specgen/generate/kube/kube.go [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+	}
+	if containerYAML.SecurityContext.Privileged != nil {
+		s.Privileged = *containerYAML.SecurityContext.Privileged
+	}
+
+	if containerYAML.SecurityContext.AllowPrivilegeEscalation != nil {
+		s.NoNewPrivileges = !*containerYAML.SecurityContext.AllowPrivilegeEscalation
+	}
+
+	if seopt := containerYAML.SecurityContext.SELinuxOptions; seopt != nil {
```
Comment (mheon) [inline pkg/specgen/generate/kube/kube.go:203]: @rhatdan PTAL at this bit

File: pkg/specgen/generate/kube/kube.go (hunk 3 above)
Comment (baude) [reply pkg/specgen/generate/kube/kube.go:119]: it isnt, i tested

File: pkg/specgen/generate/kube/kube.go (hunk 4 above)
Comment (baude) [reply pkg/specgen/generate/kube/kube.go:175]: it does not appear to need to be

File: pkg/specgen/generate/kube/kube.go (hunk 5 above)
Comment (baude) [reply pkg/specgen/generate/kube/kube.go:203]: he confirmed this

**Comment by baude** [comment]
looks like i have to move around the unit tests

File: pkg/specgen/generate/kube/kube.go (hunk 3 above)
Comment (mheon) [reply pkg/specgen/generate/kube/kube.go:119]: Well that sounds like a bug in SpecGen

**Comment by rhatdan** [comment]
Fixes: https://github.com/containers/podman/issues/8211

File: pkg/specgen/generate/kube/kube.go (hunk 1 above)
Comment (baude) [reply pkg/specgen/generate/kube/kube.go:23]: given this is only a code migration, I am going to not fix todos in the code that existed prior.

**Comment by baude** [comment]
merge me

**Comment by mheon** [comment]
LGTM
@containers/podman-maintainers PTAL

**Comment by mheon** [comment]
They really need to put the close and comment buttons a little further apart

File: pkg/domain/infra/abi/play_test.go [hunk 6]
```diff
@@ -6,34 +6,9 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	v1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (haircommander) [inline pkg/domain/infra/abi/play_test.go:9]: I'm not sure I love this change, why did you do it?

**Comment by haircommander** [comment]
one note, not blocking, otherwise LGTM

File: pkg/domain/infra/abi/play_test.go (hunk 6 above)
Comment (baude) [reply pkg/domain/infra/abi/play_test.go:9]: whats the concern? the new alias?

Comment (haircommander) [reply pkg/domain/infra/abi/play_test.go:9]: yeah! v12 doesn't mean very much to me, whereas metav1 (despite not having a ton of meaning) is a little more clear

**Pull request merged**